packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -191,3 +195,41 @@ export const setStrategyAvailability = (
 
   apiRef.current.unstable_setStrategyAvailability('rowTree', ROW_GROUPING_STRATEGY, isAvailable);
 };
+
+export const getGroupingRules = ({
+  sanitizedRowGroupingModel,
+  columnsLookup,
+}: {
+  sanitizedRowGroupingModel: GridRowGroupingModel;
+  columnsLookup: GridColumnRawLookup;
+}): GridGroupingRules =>
+  sanitizedRowGroupingModel.map((field) => ({
+    field,
+    groupingValueGetter: columnsLookup[field]?.groupingValueGetter,
+  }));
+
+/**
+ * Compares two sets of grouping rules to determine if they are equal or not.
+ */
+export const hasGroupingRulesChanged = (
+  previousValue: GridGroupingRules | undefined = [],
+  newValue: GridGroupingRules,
+) => {
+  if (previousValue.length !== newValue.length) {
+    return true;
+  }
+
+  return newValue.some((newRule, newRuleIndex) => {
+    const previousRule = previousValue?.[newRuleIndex];
+
+    if (previousRule?.groupingValueGetter !== newRule?.groupingValueGetter) {
+      return true;
+    }
+
+    if (previousRule?.field !== newRule?.field) {
```

Comment:
Not sure we need that optional chaining here
```suggestion
    const previousRule = previousValue[newRuleIndex];

    if (previousRule.groupingValueGetter !== newRule.groupingValueGetter) {
      return true;
    }

    if (previousRule.field !== newRule.field) {
```